frontend/packages/console-shared/src/components/synced-editor/index.tsx

```diff
@@ -34,14 +40,36 @@ export const SyncedEditor: React.FC<SyncedEditorProps> = ({
   onChange = _.noop,
   prune,
   YAMLEditor,
+  lastViewedEditorTypeUserSettingKey,
 }) => {
   const { formContext, yamlContext } = context;
   const { t } = useTranslation();
   const [formData, setFormData] = React.useState<K8sResourceKind>(initialData);
   const [yaml, setYAML] = React.useState(safeJSToYAML(initialData));
-  const [type, setType] = React.useState<EditorType>(initialType);
   const [safeToSwitch, setSafeToSwitch] = React.useState<boolean>(true);
   const [yamlWarning, setYAMLWarning] = React.useState<boolean>(false);
+  const [
+    lastViewedEditorType,
+    setLastViewedEditorType,
+    lastViewedEditorTypeLoaded,
+  ] = useUserSettings<EditorType>(lastViewedEditorTypeUserSettingKey);
+  const [preferredEditorType, preferredEditorTypeLoaded] = usePreferredCreateEditMethod();
+
+  const getEditorType = (): EditorType => {
+    if (
+      preferredEditorType &&
+      preferredEditorType !== PREFERRED_CREATE_EDIT_METHOD_USER_SETTING_VALUE_LATEST
+    ) {
+      return preferredEditorType as EditorType;
+    }
+    if (lastViewedEditorType) {
+      return lastViewedEditorType;
+    }
+    return initialType;
+  };
+  const [type, setType] = React.useState<EditorType>(getEditorType());
+
+  const loaded = preferredEditorTypeLoaded || lastViewedEditorTypeLoaded;
```

Comment:
Seems like most of this can be extracted out into a separate hook since its being used in `SyncedEditorField` as well.

Reply:
as per discussion, updated to a custom hook.